plugins/module_utils/gcp_utils.py

```diff
@@ -246,8 +246,9 @@ def _credentials(self):
             return svc_acct_creds.with_scopes(self.module.params['scopes'])
 
         if cred_type == 'machineaccount':
-            return google.auth.compute_engine.Credentials(
-                self.module.params['service_account_email'])
+            email = self.module.params['service_account_email']
+            email = email if email is not None else "default"
```

Comment:
These two `email =` lines can just be one line:
```python
email = self.module.params.get('service_account_email', 'default')
```
or even
```python
email = self.module.params.get('service_account_email')
```
because when you construct `Credentials(None)` you get the default service account anyway.